src/handle_locator/locate_handle.cpp

```diff
@@ -0,0 +1,211 @@
+#include <ros/ros.h>
+#include <actionlib/server/simple_action_server.h>
+#include <tue_msgs/LocateDoorHandleAction.h>
+#include <pcl_ros/transforms.h>
+#include <pcl_conversions/pcl_conversions.h>
+
+#include <pcl/filters/passthrough.h>
+#include <pcl/filters/voxel_grid.h>
+#include <pcl/filters/extract_indices.h>
+#include <pcl/search/kdtree.h>
+#include <pcl/segmentation/sac_segmentation.h>
+#include <pcl/segmentation/extract_clusters.h>
+#include <pcl/common/common.h>
+
+#include <pcl_ros/point_cloud.h>
+#include <pcl/point_types.h>
+
+typedef pcl::PointCloud<pcl::PointXYZ> PointCloud;
+
+actionlib::SimpleActionServer<tue_msgs::LocateDoorHandleAction>* as_;
+std::shared_ptr<tf::TransformListener> listener_;
+
+geometry_msgs::PointStamped handle_edge_point1;
+geometry_msgs::PointStamped handle_edge_point2;
+
+void executeCB(const tue_msgs::LocateDoorHandleGoalConstPtr& goal) {
```

Comment:
I would suggest cutting this monolithic function up into smaller functional components. To make it better readable and testable. Such as the cropping of the pointcloud and the calculating of the error